pkg/agent/openflow/pipeline.go

```diff
@@ -551,11 +586,15 @@ func (c *client) defaultFlows() (flows []binding.Flow) {
 
 // tunnelClassifierFlow generates the flow to mark traffic comes from the tunnelOFPort.
 func (c *client) tunnelClassifierFlow(tunnelOFPort uint32, category cookie.Category) binding.Flow {
+	nextTable := conntrackTable
+	if c.enableProxyFull {
+		nextTable = serviceConntrackTable
```

Comment:
> Can we use hairpinIP as the SNAT IP? The IP can represent all hairpin cases that we want to ensure response are transmitted back to OVS.

IMO, if we don't change the existing design of client from pod hairpin Service, we cannot use hairpinIP as the SNAT IP. When dst_ip is hairpinIP, it will be first processed at table serviceHairpinTable as below.  As we can see, the destination IP will be changed, this is not we wanted.
```
table=23, priority=200,ip,nw_dst=169.254.169.252 actions=move:NXM_OF_IP_SRC[]->NXM_OF_IP_DST[],load:0x1->NXM_NX_REG0[18],resubmit(,24)
```
Another way is that we change the SNAT way of client from pod hairpin Service. When a client from pod accesses a Service whose Endpoint is the pod, the SNAT is also implemented through ct_zone. Then we can unify the hairpinIP as one. But this will introduce extra overhead for this case. Existing SNAT is implemented by stateless SNAT(just modifying     IP address, without ct)

Reply:
> Why the next table of tunnel traffic should be serviceConntrackTable?

The flow in table `serviceConntrackTable` is used to transformed response SNATed Service packetes. If going to table `conntrackTable` directly, the response Service packets cannot be transformed correctly.